openedx/core/djangoapps/user_authn/views/register.py

```diff
@@ -170,9 +167,14 @@ def create_account_with_params(request, params):
     third_party_auth_credentials_in_api = 'provider' in params
     is_third_party_auth_enabled = third_party_auth.is_enabled()
 
-    if is_third_party_auth_enabled and (pipeline.running(request) or third_party_auth_credentials_in_api):
+    is_sso = is_sso_request(request)
+    if is_sso:
         params["password"] = generate_password()
 
+    if is_registration_api_v1(request) or is_sso:
+        if 'confirm_email' in extra_fields:
```

Comment:
while testing this, I found (might be relateable)
If a user signups by using **Gmail SSO** with email **test@yopmail.com** and when he lands on the registration page where we need some manual information from the user and he enters **test123@yopmail.com** as his email then it is accepted without any confirmation.
I am attaching the screenshot of the page I am talking about.
<img width="1416" alt="Screenshot 2020-07-01 at 12 44 56 PM" src="https://user-images.githubusercontent.com/11922730/86218272-fc916980-bb99-11ea-9e51-0cee7bc75108.png">

Reply:
It seems like it is not related to this change so I will discuss with team and create a new ticket for this.